tests/Sulu/Component/Rest/ListBuilder/Metadata/FieldDescriptorFactoryTest.php

```diff
@@ -0,0 +1,201 @@
+        );
+        $parameterBag->resolveValue(Argument::any())->willReturnArgument(0);
+
+        $configCache = $this->prophesize(ConfigCache::class);
+        $configCache->isFresh()->willReturn(false);
+        $configCache->write(Argument::any(), null)->willReturn(null);
+
+        $chain = [
+            new MetadataProvider(
+                new MetadataFactory(new DoctrineXmlDriver($locator->reveal(), $parameterBag->reveal()))
+            ),
+            new MetadataProvider(new MetadataFactory(new GeneralXmlDriver($locator->reveal()))),
+        ];
+        $provider = new ChainProvider($chain);
+        $factory = new FieldDescriptorFactory($provider, $configCache->reveal());
+        $fieldDescriptor = $factory->getFieldDescriptorForClass(\stdClass::class);
+
+        self::assertEquals(['id', 'firstName', 'lastName'], array_keys($fieldDescriptor));
+
+        $this->fieldDescriptorTest($fieldDescriptor['id'], 'id', 'public.id', true, false, 'integer');
+        $this->fieldDescriptorTest(
+            $fieldDescriptor['firstName'],
+            'firstName',
+            'contact.contacts.firstName',
+            false,
+            true
+        );
+        $this->fieldDescriptorTest($fieldDescriptor['lastName'], 'lastName', 'contact.contacts.lastName', false, true);
+    }
+
+    public function testGetFieldDescriptorForClassEmpty()
+    {
+        $locator = $this->prophesize(FileLocatorInterface::class);
+        $locator->findFileForClass(new \ReflectionClass(new \stdClass()), 'xml')
+            ->willReturn(__DIR__ . '/Resources/empty.xml');
+
+        $parameterBag = $this->prophesize(ParameterBagInterface::class);
+        $parameterBag->resolveValue('%sulu.model.contact.class%')->willReturn('SuluContactBundle:Contact');
+        $parameterBag->resolveValue('%sulu.model.contact.class%.avatar')->willReturn(
+            'SuluContactBundle:Contact.avatar'
+        );
+        $parameterBag->resolveValue('%sulu.model.contact.class%.contactAddresses')->willReturn(
+            'SuluContactBundle:Contact.contactAddresses'
+        );
+        $parameterBag->resolveValue(Argument::any())->willReturnArgument(0);
+
+        $configCache = $this->prophesize(ConfigCache::class);
+        $configCache->isFresh()->willReturn(false);
+        $configCache->write(Argument::any(), null)->willReturn(null);
+
+        $chain = [
+            new MetadataProvider(
+                new MetadataFactory(new DoctrineXmlDriver($locator->reveal(), $parameterBag->reveal()))
+            ),
+            new MetadataProvider(new MetadataFactory(new GeneralXmlDriver($locator->reveal()))),
+        ];
+        $provider = new ChainProvider($chain);
+        $factory = new FieldDescriptorFactory($provider, $configCache->reveal());
+        $fieldDescriptor = $factory->getFieldDescriptorForClass(\stdClass::class);
+
+        self::assertEmpty($fieldDescriptor);
+    }
+
+    protected function fieldDescriptorTest(
```

Comment:
`s/field/assertField/g`. If we write custom tests they should be prefixed with `assert` as with Phpunit.